src/wasm/syscalls_openmp.cpp

```diff
@@ -190,10 +208,120 @@ namespace wasm {
             }
         }
 
-        // Reset the thread count for this thread
+        // Reset the thread count for the master thread
         thisSectionThreadCount = 1;
     }
 
+    /**
+     * @param    loc       Source code location
+     * @param    gtid      Global thread id of this thread
+     * @param    schedule  Scheduling type for the parallel loop
+     * @param    plastiter Pointer to the "last iteration" flag
+     * @param    plower    Pointer to the lower bound
+     * @param    pupper    Pointer to the upper bound of loop chunk
+     * @param    pstride   Pointer to the stride for parallel loop
+     * @param    incr      Loop increment
+     * @param    chunk     The chunk size for the parallel loop
+     *
+     * The functions compute the upper and lower bounds and strides to be used for the
+     * set of iterations to be executed by the current thread from the statically
+     * scheduled loop that is described by the initial values of the bounds, strides,
+     * increment and chunks for parallel loop and distribute constructs.
+     *
+     * See sched_type for supported scheduling.
+     *
+     * SAFETY: The implementation here is not thread safe at the moment. The pointers values, although set
+     * to the correct values for each thread right before calling this function (on native, COW, so this
+     * function's modifications to them should be thread-local but are not at the moment). This function will
+     * therefore only work if each thread's work between init and fini is mutually exclusive..
+     * TODO: Find a way to make this thread safe, the hard bit is that the compiler assumes that those
+     * variables, although at the same address, will be thread-local.
```

Comment:
@Shillaker Have you encountered something similar before? I can't see a way to do this without modifying the compiler or WAVM. I need those 4 variables to follow the same semantic as `thread_local`